From ed45e0d6c48f361ed9d81c18657de3b5fedca458 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 9 Oct 2017 17:27:21 +0300
Subject: [PATCH 2025/2241] arm64: defconfig: disable features in PD defconfig
 to reduce boot time

Disable features in PD defconfig to reduce boot time in emilation,
and to reduce kernel image size

Change-Id: I5c739ade80001eec8c4c0ba52d7cc9370dfcdfcd
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/45083
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
---
 arch/arm64/configs/mvebu_v8_pd_lsp_defconfig | 47 ++++++----------------------
 1 file changed, 9 insertions(+), 38 deletions(-)

diff --git a/arch/arm64/configs/mvebu_v8_pd_lsp_defconfig b/arch/arm64/configs/mvebu_v8_pd_lsp_defconfig
index 3b28b0f..fc69159 100644
--- a/arch/arm64/configs/mvebu_v8_pd_lsp_defconfig
+++ b/arch/arm64/configs/mvebu_v8_pd_lsp_defconfig
@@ -26,18 +26,16 @@ CONFIG_ARCH_MVEBU_PD=y
 CONFIG_PCI=y
 CONFIG_PCI_MSI=y
 CONFIG_SCHED_MC=y
-CONFIG_HOTPLUG_CPU=y
 CONFIG_PREEMPT=y
 CONFIG_HZ_20=y
 CONFIG_KSM=y
 CONFIG_TRANSPARENT_HUGEPAGE=y
 CONFIG_CMA=y
 CONFIG_CMDLINE="console=ttyAMA0"
+# CONFIG_EFI is not set
 # CONFIG_CORE_DUMP_DEFAULT_ELF_HEADERS is not set
 CONFIG_COMPAT=y
 # CONFIG_SUSPEND is not set
-CONFIG_NET=y
-CONFIG_INET=y
 CONFIG_UEVENT_HELPER_PATH="/sbin/hotplug"
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
@@ -47,9 +45,10 @@ CONFIG_BLK_DEV_LOOP=y
 CONFIG_BLK_DEV_RAM=y
 CONFIG_BLK_DEV_RAM_COUNT=1
 CONFIG_BLK_DEV_RAM_SIZE=48000
-CONFIG_SCSI=y
-# CONFIG_SCSI_PROC_FS is not set
-# CONFIG_SCSI_LOWLEVEL is not set
+CONFIG_MD=y
+CONFIG_BLK_DEV_MD=y
+# CONFIG_MD_AUTODETECT is not set
+CONFIG_MD_RAID456=y
 # CONFIG_INPUT_MOUSEDEV is not set
 CONFIG_INPUT_EVDEV=y
 # CONFIG_INPUT_KEYBOARD is not set
@@ -61,43 +60,29 @@ CONFIG_LEGACY_PTY_COUNT=16
 CONFIG_SERIAL_8250=y
 CONFIG_SERIAL_8250_CONSOLE=y
 CONFIG_SERIAL_8250_DW=y
-CONFIG_SERIAL_AMBA_PL011=y
-CONFIG_SERIAL_AMBA_PL011_CONSOLE=y
-CONFIG_SERIAL_OF_PLATFORM=y
-CONFIG_SERIAL_XILINX_PS_UART=y
-CONFIG_SERIAL_XILINX_PS_UART_CONSOLE=y
-CONFIG_SERIAL_MVEBU_UART=y
 # CONFIG_HW_RANDOM is not set
-CONFIG_I2C=y
-CONFIG_I2C_CHARDEV=y
-CONFIG_I2C_MV64XXX=y
 CONFIG_PPS=y
 # CONFIG_HWMON is not set
-CONFIG_MFD_SYSCON=y
+CONFIG_REGULATOR=y
+# CONFIG_HID is not set
 # CONFIG_USB_SUPPORT is not set
 CONFIG_DMADEVICES=y
-CONFIG_MV_XOR=y
 CONFIG_MV_XOR_V2=y
 CONFIG_DMATEST=y
 CONFIG_COMMON_CLK_VERSATILE=y
 CONFIG_CLK_SP810=y
 CONFIG_CLK_VEXPRESS_OSC=y
 # CONFIG_IOMMU_SUPPORT is not set
-CONFIG_RESET_CONTROLLER=y
-CONFIG_RAS=y
-# CONFIG_DMIID is not set
+# CONFIG_PHONE is not set
 CONFIG_EXT2_FS=y
 CONFIG_EXT2_FS_XATTR=y
 CONFIG_EXT3_FS=y
 CONFIG_FANOTIFY=y
-CONFIG_FANOTIFY_ACCESS_PERMISSIONS=y
 CONFIG_QUOTA=y
 CONFIG_AUTOFS4_FS=y
-CONFIG_FUSE_FS=y
-CONFIG_CUSE=y
 CONFIG_TMPFS=y
-# CONFIG_EFIVAR_FS is not set
 # CONFIG_MISC_FILESYSTEMS is not set
+CONFIG_NLS=y
 CONFIG_NLS_CODEPAGE_437=y
 CONFIG_NLS_ISO8859_1=y
 CONFIG_PRINTK_TIME=y
@@ -110,17 +95,3 @@ CONFIG_LOCKUP_DETECTOR=y
 # CONFIG_DEBUG_PREEMPT is not set
 # CONFIG_FTRACE is not set
 CONFIG_MEMTEST=y
-CONFIG_KEYS=y
-CONFIG_SECURITY=y
-CONFIG_CRYPTO_ANSI_CPRNG=y
-CONFIG_ARM64_CRYPTO=y
-CONFIG_CRYPTO_SHA1_ARM64_CE=y
-CONFIG_CRYPTO_SHA2_ARM64_CE=y
-CONFIG_CRYPTO_GHASH_ARM64_CE=y
-CONFIG_CRYPTO_AES_ARM64_CE_CCM=y
-CONFIG_CRYPTO_AES_ARM64_CE_BLK=y
-CONFIG_CRYPTO_AES_ARM64_NEON_BLK=y
-CONFIG_CRYPTO_CRC32_ARM64=y
-CONFIG_CRC_ITU_T=y
-CONFIG_CRC7=y
-CONFIG_LIBCRC32C=y
-- 
2.7.4

